webapp/src/services/webSocket/types.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2024, RTE (https://www.rte-france.com)
+ *
+ * See AUTHORS.txt
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This file is part of the Antares project.
+ */
+
+import type {
+  CommandResultDTO,
+  GenericInfo,
+  LaunchJobDTO,
+  LaunchJobProgressDTO,
+  StudyMetadata,
+} from "@/common/types";
+import { WsEventType } from "./constants";
+import { O } from "ts-toolbelt";
+import { FileDownloadDTO } from "../api/downloads";
+import { TaskDTO, TTaskType } from "../api/tasks/types";
+
+/**
+ * Copyright (c) 2024, RTE (https://www.rte-france.com)
+ *
+ * See AUTHORS.txt
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This file is part of the Antares project.
+ */
+
+export type TWsEventType = O.UnionOf<typeof WsEventType>;
```

Comment:
same here I personally find the suffix T confusing, maybe we should use prefix for all derived types what do you think ?

Reply:
See my comment below, in fact `T` suffix is a common convention, personally I use it when I the name I want is already used.